MaterialLib/MPL/Properties/RelPermLiakopoulos.cpp

```diff
@@ -0,0 +1,98 @@
+/**
+ * \file
+ * \author Norbert Grunwald
+ * \brief
+ *
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "MaterialLib/MPL/Properties/RelPermLiakopoulos.h"
+#include "MaterialLib/MPL/Medium.h"
+
+#include <algorithm>
+#include <cmath>
+
+namespace MaterialPropertyLib
+{
+/**
+ */
+PropertyDataType RelPermLiakopoulos::value(
+    VariableArray const& variable_array,
+    ParameterLib::SpatialPosition const& pos,
+    double const t) const
+{
+    /// here, an extra computation of saturation is forced, guaranteeing a
+    /// correct value. In order to speed up the computing time, saturation could
+    /// be insertred into the primary variable array after it is computed in the
+    /// FEM assembly.
+    auto const s_L = _medium->property(PropertyType::saturation)
+                         .template value<double>(variable_array, pos, t);
+    auto const s_L_res = _residual_liquid_saturation;
+    auto const k_rel_min_GR = _min_relative_permeability_gas;
+
+    if (s_L <= s_L_res)
```

Comment:
Here, you do not want to restrict s_L. Don't you?

Reply:
I'm not sure what you mean. If I don't restrict it, I obtain a negative effective saturation for `s_L < s_L_res` and can't solve for `k_rel_GR`... Did I miss something?